test/integration/api/forms/forms.js

```diff
@@ -74,8 +74,12 @@ describe('api: /projects/:id/forms (create, read, update)', () => {
             .set('Content-Type', 'application/xml')
             .expect(409)
             .then(({ body }) => {
-              body.details.fields.should.eql([ 'projectId', 'xmlFormId', 'version' ]);
-              body.details.values.should.eql([ '1', 'simple', '' ]);
+              // eslint-disable-next-line no-multi-str
+              body.message.should.eql("You tried to publish the form 'simple' with version '', \
```

Comment:
The error response used to look like
```
body {
  message: 'A resource already exists with projectId,xmlFormId,version value(s) of 1,simple,.',
  code: 409.3,
  details: {
    fields: [ 'projectId', 'xmlFormId', 'version' ],
    values: [ '1', 'simple', '' ],
    table: 'forms'
  }
}
```

Because the form database insert also included setting publishedAt, and that violated a constraint. 

Now the constraint is violated at a different time in a different way. The message is in some ways clearer and in other ways wrong because it's referring to the .../formid/draft/publish endpoint.

Reply:
One thing you could do is remove the last one or two sentences of the 409.6 (`Problem.user.uniquenessViolation`) message. I think the resulting message would work for both endpoints. I'm pretty sure that Frontend shows its own message for a 409.6 in the `FormDraftPublish` component, so a change to the Backend message would only affect API users.

Alternatively, in `Forms.publish()`, you could prevent the translation of `Problem.user.uniquenessViolation` to `Problem.user.versionUniquenessViolation` when `sameAsCreate` is `true`. That should result in the same behavior as before.

Another option: you could pass `sameAsCreate` to `Problem.user.versionUniquenessViolation()`, which could then return a different message depending on whether `sameAsCreate` is `true` or `false`.

Reply:
I like the idea of just removing the end of the error message.